Russian State Library.js

```diff
@@ -0,0 +1,1558 @@
+	https://favorites.rsl.ru/(ru|en)/
+*/
+const sRSLFilters = {
+	libraryCatalog: "Российская Государственная Библиотека",
+	marcTableCSS: "div#marc-rec > table",
+	descTableCSS: "table.card-descr-table",
+	searchListCSS: "span.js-item-maininfo",
+	searchRecordRslidAttr: "data-id",
+	searchRecordTitle: /^[^:/[]*/,
+	searchPattern: "https://search.rsl.ru/ru/search#q=id:{@id@} AND title:({@title@})",
+	rslidPrefix: "https://search.rsl.ru/ru/record/",
+	thesisRelAttr: "href",
+	thesisRelPrefix: "/ru/transition/",
+	thesisRelCSS: 'a[href^="/ru/transition/"]',
+	favRslidCSS: "a.rsl-link",
+	favDescCSS: "div.rsl-fav-item-descr",
+	title: "Заглавие",
+	catalog: "Каталоги",
+	bbk: "BBK-код",
+	callNumber: "Места хранения",
+	eResource: "Электронный адрес"
+};
+
+/*
+	Filter strings for extraction of metadata from
+	aleph.rsl.ru
+*/
+const aRSLFilters = {
+	marcTableTagCSS: "td.td1[nowrap]",
+	marcTableValCSS: "td.td1:not([nowrap])",
+	marcTableSetCSS: 'a[title="Добавить в подборку"]',
+	recordMarcSignature: "&format=001",
+	recordStandardSignature: "&format=999",
+	recordFormatRegex: /&format=[0-9]{3}/,
+	urlPrefix: "http://aleph.rsl.ru/F/",
+};
+
+
+const baseEurl = 'https://dlib.rsl.ru/';
+
+
+function attr(docOrElem, selector, attr, index) {
+	let elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector);
+	return elem ? elem.getAttribute(attr) : null;
+}
+
+
+/**
+ *	Adds link attachment to a Zotero item.
+ *
+ *  Note attachment creation issue.
+ *  Problem manifestation:
+ *  If 'linkMode: "linked_url"' is not present or is the last (possibly also in
+ *  the middle) line in the object definition, the "title" property is ignored,
+ *  e.g., for 'research.rsl.ru/ru/record/01004956040' the "title" is set to
+ *  "01004956040" regardless of the specified title. If 'linkMode: "linked_url"'
+ *  is the first member as below, the "title" field is set as expected.
+ *
+ *  There appear to be a bug in Zotero routine that creates attachments from JSON
+ *  definitions. "linkMode" property affects (determines?) the type of the new
+ *  attachment and interpretation of the remaining properties. The routine
+ *  apparently enumerates keys of the supplied JSON object, instead of accessing
+ *  "linkMode" directly. It processes properties in the order supplied, and if
+ *  "linkMode" does not come first, may not process earlier properties correctly.
+ *  This is particularly problematic, since there is no guaranteed order of members
+ *  in a dictionary, though, apparently, in simple cases at least the members are
+ *  returned in the "added first" order. However, this behavior is not relaibly
+ *  reproducible.
+ *  Windows 7 x64, April 2020.
+ *
+ *	@param {Object} item - Zotero item
+ *	@param {String} title - Link name
+ *	@param {String} url - Link url
+ *
+ *	@return {None}
+ */
+function addLink(item, title, url) {
+	item.attachments.push({ linkMode: "linked_url", // Apparently, should be the first
```

Comment:
I'm a bit confused by this -- `linkMode` should not be set in web translators nor should it be necessary. How are you seeing the issue you mention above manifest? We have many translators that just save links using `snapshot: false` and they save/display the title properly.

Reply:
Basically, without `linkMode` , the specified link name was ignored, as if it was not specified at all. This behavior was not reliably reproducible though. When I tried to run tests later without `linkMode`, the specified link name was set as expected. However, I also observed the same issue when I tried to create link attachments via JS API in the built-in JS console.